frontend/simple/js/transactions.js

```diff
@@ -0,0 +1,263 @@
+  }
+  async load () {
+    if (_isLoaded.get(this)) { return }
+    try {
+      let transactionID = await transactionDB.getItem('nextID')
+      if (!transactionID) {
+        transactionID = 0
+        await transactionDB.setItem('nextID', transactionID)
+      } else {
+        await _transactionID.set(this, transactionID)
+      }
+      let failures = await transactionDB.getItem('failures') || []
+      _failures.set(this, failures)
+      for (let failed of failures) {
+        let json = await transactionDB.getItem(`${failed.id}`)
+        let transaction = Transaction.fromJSON(json)
+        _transactionID.set(transaction, failed.id)
+        this.enqueue(transaction)
+      }
+      this.run()
+    } catch (ex) {
+      console.log(ex)
+    }
+  }
+
+  enqueue (transaction) {
+    let transactionQueue = _transactionQueue.get(this)
+    transactionQueue.push(transaction)
+  }
+
+  async run (transaction) {
+    if (transaction) {
+      this.enqueue(transaction)
+    }
+    if (!_isRunning.get(this)) {
+      _isRunning.set(this, true)
+      let transactionQueue = _transactionQueue.get(this)
+      let next = transactionQueue.pop()
+      while (next) {
+        // Establish a unique identifier for each transaction
+        if (!_transactionID.has(next)) {
+          let transactionID = _transactionID.get(this)
+          // Set transaction's ID and increment The next ID
+          _transactionID.set(next, transactionID)
+          _transactionID.set(this, transactionID + 1)
+          await transactionDB.setItem('nextID', _transactionID.get(this))
+        }
+        try {
+          await next.run()
+          let backup = await transactionDB.getItem(`${_transactionID.get(next)}`)
+          if (backup) {
+            let failures = _failures.get(this)
+            let index = failures.findIndex(failure => failure.id === _transactionID.get(next))
+            if (index > -1) {
+              failures.splice(index, 1)
+              await transactionDB.setItem('failures', failures)
+            }
+            await transactionDB.removeItem(`${_transactionID.get(next)}`)
+          }
+        } catch (ex) {
+          if (next.isRecoverable) {
+            // Save a serialized version of the transaction
+            let transactionID = _transactionID.get(next)
+            let failures = _failures.get(this)
+            let failure = failures.find(failure => failure.id === transactionID)
+            if (!failure) {
+              failure = { id: transactionID, retryCount: 0 }
+              failures.push(failure)
+              await transactionDB.setItem('failures', failures)
+            } else {
+              failure.retryCount++
+              await transactionDB.setItem('failures', failures)
+            }
+            await transactionDB.setItem(`${transactionID}`, next.toJSON())
+            let retry = next
+            setTimeout(() => {
+              this.run(retry)
+            }, 500 * fibSequence(failure.retryCount))
```

Comment:
We need to discuss this, as it's extremely unlikely that even an external failure should indefinitely keep trying like this.